pkg/sdn/plugin/proxy.go

```diff
@@ -49,15 +50,19 @@ func NewProxyPlugin(pluginName string, osClient *osclient.Client, kClient *kclie
 	}
 
 	return &OsdnProxy{
-		kClient:  kClient,
-		osClient: osClient,
-		ids:      make(map[string]uint32),
-		firewall: make(map[string]*proxyFirewallItem),
+		kClient:   kClient,
+		osClient:  osClient,
+		ids:       make(map[string]uint32),
+		egressDNS: &EgressDNS{},
+		firewall:  make(map[string]*proxyFirewallItem),
 	}, nil
 }
 
 func (proxy *OsdnProxy) Start(baseHandler pconfig.EndpointsConfigHandler) error {
 	glog.Infof("Starting multitenant SDN proxy endpoint filter")
+	if err := CheckDNSResolver(); err != nil {
```

Comment:
Same question here as @danwinship asked in the previous commit; do we want to fail the node completely, rather than just make egress DNS not work?  This might catch people by surprise, especially when they don't use egress DNS at all.